src/encoded/static/components/matrix_reference_epigenome.js

```diff
@@ -0,0 +1,715 @@
+        this.hasRequestedClassifications = filteredClassifications.length > 0;
+    }
+
+    componentDidMount() {
+        this.handleScrollIndicator(this.scrollElement);
+
+        // Display the organism-chooser modal if the query string doesn't specify an organism to
+        // display.
+        this.setState({ organismChooserVisible: this.initialSelectedTab === null });
+    }
+
+    componentDidUpdate() {
+        // Updates only happen for scrolling on this page. Every other update causes an
+        // unmount/mount sequence.
+        this.handleScrollIndicator(this.scrollElement);
+    }
+
+    /**
+     * Get a list of organism scientific names that exist in the given matrix data.
+     * @return {array} Organisms in data; empty array if none
+     */
+    getAvailableOrganisms() {
+        const { context } = this.props;
+        const organismFacet = context.facets && context.facets.find(facet => facet.field === 'replicates.library.biosample.donor.organism.scientific_name');
+        if (organismFacet) {
+            return organismFacet.terms.map(term => term.key);
+        }
+        return [];
+    }
+
+    /**
+     * Get an array of all values in the current query string corresponding to the given `key`.
+     * @param {string} key Query string key whose values this retrieves.
+     *
+     * @return {array} Values for each occurence of `key` in the query string.
+     */
+    getQueryValues(key) {
+        return this.query.getKeyValues(key);
+    }
+
+    /**
+     * Generate React components to render the tabs with <TabPanel>. These components correspond
+     * to the organisms available in the given matrix data.
+     *
+     * @return {object} React components for each tab; null if no organisms
+     */
+    getOrganismTabs() {
+        const organisms = this.getAvailableOrganisms();
+        const organismTabs = {};
+        if (organisms.length > 0) {
+            organisms.forEach((organismName) => {
+                organismTabs[organismName] = <i>{organismName}</i>;
+            });
+            return organismTabs;
+        }
+        return null;
+    }
+
+    /**
+     * Get the ID of the organism tab that should be selected on page load. With no organism
+     * specified in the query string, null gets returned which causes the organism-selector modal
+     * to appear.
+     *
+     * @return {string} ID of the tab that should be selected; null if none specified
+     */
+    getInitialSelectedTab() {
+        const organisms = this.getAvailableOrganisms();
+        const selectedOrganisms = this.getQueryValues('replicates.library.biosample.donor.organism.scientific_name');
+        if (selectedOrganisms.length === 1) {
+            // Query string specifies exactly one organism. Select the corresponding tab if it
+            // exists, otherwise don't select a tab.
+            return organisms.includes(selectedOrganisms[0]) ? selectedOrganisms[0] : null;
+        }
+        return null;
```

Comment:
I think this was to avoid rewriting the URL for what’s likely to be a situation that never happens, but this might be more of a question for @hitz and @gabdank.